open_spiel/games/gin_rummy.cc

```diff
@@ -665,6 +771,28 @@ GinRummyGame::GinRummyGame(const GameParameters& params)
       undercut_bonus_(ParameterValue<int>("undercut_bonus")) {
   SPIEL_CHECK_GE(knock_card_, 0);
   SPIEL_CHECK_LE(knock_card_, kDefaultKnockCard);
+  default_observer_ = std::make_shared<GinRummyObserver>(kDefaultObsType);
+  // TODO(Michal) Should we implement perfect recall?
+  //info_state_observer_ = std::make_shared<GinRummyObserver>(kInfoStateObsType);
+}
+
+// TODO(Michal) This is how it's implemented in dark chess. The commented out
```

Comment:
I think the observer registration in Leduc is something (relatively) new, I am not sure what is the intended usage... Maybe @elkhrt can explain?

Reply:
I removed the commented out code and followed the implementation in dark chess. If we encounter a use case for the observer registration I'll be happy to add it in.